### ISSUE TITLE
Still getting "Error: Cannot mark location in editor for (no-useless-escape)"

### ISSUE THREAD

**Issue by callumlocke** [AtomLinter/linter-eslint#743]
## Issue Type

Bug
## Issue Description

![image](https://cloud.githubusercontent.com/assets/250617/19726003/a697476c-9b80-11e6-8fa0-bc9d9ea22fbd.png)

I realise this may have been reported and marked resolved before (#700), but I'm still getting this error.

Here's what I have done just now:
- uninstalled atom
- deleted my `~/atom` directory
- deleted the `node_modules` directory from my project
- restarted OS X
- ran `npm install` in my project
- installed a fresh copy of atom and opened it
- installed these packages:
  - linter (1.11.18)
  - linter-eslint (8.0.0)

When I open a JS file, I still get the error.

I do not have a global eslint installed. I just use local modules. My local eslint is v3.8.1.

```
Error: Cannot mark location in editor for (no-useless-escape) - (Unnecessary escape character: \/.) at line (102) column (52)
Error: Cannot mark location in editor for (no-useless-escape) - (Unnecessary escape character: \/.) at line (102) column (52)
    at /Users/callum/.atom/packages/linter-eslint/lib/main.js:227:21
    at Array.map (native)
    at /Users/callum/.atom/packages/linter-eslint/lib/main.js:192:27
```

**Comment by IanVS** [comment]
Thanks, this looks like a new issue.  It's possible that ESLint is reporting an incorrect position for that rule (`no-useless-escape`).  Is it possible for you to provide the line of text that is occurring on, or another reproducible line?

**Comment by jstahlmann** [comment]
I'm having the same issue!
![bildschirmfoto 2016-10-27 um 14 16 29](https://cloud.githubusercontent.com/assets/8123408/19766832/3cf54568-9c50-11e6-8c15-ac7efff6aff2.png)

**Comment by IanVS** [comment]
@jstahlmann yours is a different issue, notice the different rule name).  Upgrade to ESLint 3.5.0 or above and that will be fixed.

**Comment by mc-funk** [comment]
I am having the same issue as @jstahlmann with v8.0.0 and the Atom 1.12.1:
<img width="1120" alt="screen shot 2016-11-14 at 3 15 19 pm" src="https://cloud.githubusercontent.com/assets/10001418/20283102/4428921e-aa7d-11e6-8824-6fd4e03f53d2.png">
I believe I have had it with other linting errors too, but I haven't reproduced them lately.
Can make a new ticket, but wanted to make sure on this one to corroborrate @jstahlmann 's issue as one that happens with 8.0.0 .

**Comment by IanVS** [comment]
Again, this is not related to the version of `linter-eslint`, but rather the version of `eslint` itself. You will need to upgrade your version of ESLint to 3.5.0 or above, or if you are relying on the ESLint that is bundled as part of `linter-eslint`, you can try uninstalling it completely, and then re-installing it.  Restart Atom, and you should be good to go.

**Comment by mc-funk** [comment]
Thanks for the clarification; somehow I'd read through a few of these tickets and missed that distinction. 

However, it does appear that our eslint version is 3.5.2, so it seems the problem may lie somewhere else.

**Comment by IanVS** [comment]
@mc-funk can you please post the results of running `Linter Eslint: debug` from the Atom command palette?

**Comment by mc-funk** [comment]
Fascinating. I see what you mean now -- eslint in Atom is still not ^3.5. 
The following is after I uninstalled linter-eslint, restarted Atom, reinstalled, and restarted Atom again. eslint is not installed independently as a package. 

> atom version: 1.12.2 
linter-eslint version: 8.0.0
eslint version: 2.13.1
hours since last atom restart: 0
platform: darwin
Using local project eslint from /Users/michellefunk/yeti-ui/node_modules/eslint
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": []
}

**Comment by IanVS** [comment]
The key here is "Using local project eslint from /Users/michellefunk/yeti-ui/node_modules/eslint"

If you have eslint installed in the project you're working on, `linter-eslint` will use that version.  It looks like you just need to upgrade the eslint in that project (`yeti-ui`), and you should be good to go.

Or, if you can't update that version of ESLint for whatever reason, you can install eslint globally (`npm i -g eslint`), and then set the `linter-eslint` option to use the global version.  It's better to use the eslint in your project though, and global installation should be a workaround until you can update the project version.

**Comment by mc-funk** [comment]
yeti-ui was the one running 3.5.2, so now I am quite confused! 
<img width="456" alt="screen shot 2016-11-14 at 3 53 41 pm" src="https://cloud.githubusercontent.com/assets/10001418/20284410/9fdd1ada-aa82-11e6-821d-a4b13f438c66.png">

**Comment by mc-funk** [comment]
OK. Mysteries of life: I have not restarted Atom since I last got you this information. But in demonstrating to a colleague, I ran the linter-eslint debug again and got this: 

> atom version: 1.12.2
linter-eslint version: 8.0.0
eslint version: 3.9.1
hours since last atom restart: 0.6
platform: darwin
Using local project eslint from /Users/michellefunk/PSImon-game-hack/node_modules/eslint
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": []
}

Additionally, as one would expect, the multiline error is no longer present. 
Not sure what caused this oddity of debug output, but it is clear now that the uninstall/reinstall fixed the bug -- also seems that the version used is independent of what is installed in my active project.

**Comment by mc-funk** [comment]
OK so. You'll be pleased to hear that in the intervening time I have reconnected with my basic reading comprehension skills. We have some dependency issues that are keeping eslint down so I need to see if we can resolve those. Thanks for your help.

**Comment by not-an-aardvark** [comment]
It's possible that this is fixed by https://github.com/eslint/eslint/commit/d50f6c127b51f55356b4bf744c8d302dee17f96e. The `no-useless-escape` rule was reporting an invalid location under certain circumstances.

**Comment by IanVS** [comment]
Thanks @not-an-aardvark!  Looks like it was released in eslint `3.11.0`.  I'm going to close this issue, but if anyone still has the problem with eslint `3.11.0` or above, please post here and provide the output of the `Linter Eslint: Debug` command.
